### PR TITLE
Bump aca-py version to 0.6 and change API and examples accordingly - issue 56

### LINKED ISSUE #56: Migrate to ACA-Py v0.6.0
## Description
ACA-Py recently released version 0.6.0. We should bump to this but it will require some substantial refactor. This is worth it because support multi-wallet functionality.

## Are you interested in working on this improvement yourself?
- No, but I think @lohanspies or @frogman might be able to take this on. Happy to support.

## Additional Context
Add any other context or screenshots.

### PR DESCRIPTION
**Pull request by morrieinmaas**
## Description
closes #56 

## Affected Dependencies
aca-py

## How has this been tested?
- Has been manually tested in aries basic controller notebooks and the attachment notebooks.
- see repo readme and run tutorials to reproduce.
- this should really be covered with unit tests I guess.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: tutorials/aries-basic-controller/notebooks/alice/Part 2 - Aries Basic Controller.ipynb [hunk 1]
```diff
@@ -136,7 +136,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 1,
```
Comment (morrieinmaas) [inline tutorials/aries-basic-controller/notebooks/alice/Part 2 - Aries Basic Controller.ipynb:139]: will put the counter back to null. sorry i missed that change in the commit

**Comment by lohanspies** [comment]
@morrieinmaas we can extract the added AcaPy parameter from https://github.com/OpenMined/PyDentity/pull/68/files#diff-53a0dfff4019907a159c1ac9502ca216fadd8b3c2cfef9bb999e8f8fe56fdd02 into the yml file instead to keep it consistent with the rest of the repo.

**Comment by lohanspies** [comment]
> @morrieinmaas we can extract the added AcaPy parameter from https://github.com/OpenMined/PyDentity/pull/68/files#diff-53a0dfff4019907a159c1ac9502ca216fadd8b3c2cfef9bb999e8f8fe56fdd02 into the yml file instead to keep it consistent with the rest of the repo.

See you already did this in https://github.com/OpenMined/PyDentity/pull/68/files#diff-232bac0e229b355a979026366e0a706bf87b48da75db720e309ed7fa1c6b8208 and https://github.com/OpenMined/PyDentity/pull/68/files#diff-540c821638d1899c5fb9b43372d6a108952362e99666e2514c333d4d66e54a61
i.e. Not sure why it is required in the Dockerfile as well as part of the AcaPy startup cmd

**Comment by morrieinmaas** [comment]
> @morrieinmaas we can extract the added AcaPy parameter from https://github.com/OpenMined/PyDentity/pull/68/files#diff-53a0dfff4019907a159c1ac9502ca216fadd8b3c2cfef9bb999e8f8fe56fdd02 into the yml file instead to keep it consistent with the rest of the repo.

Yeah totally that makes sense. Thanks for pointing that out @lohanspies

**Comment by morrieinmaas** [comment]
> > @morrieinmaas we can extract the added AcaPy parameter from https://github.com/OpenMined/PyDentity/pull/68/files#diff-53a0dfff4019907a159c1ac9502ca216fadd8b3c2cfef9bb999e8f8fe56fdd02 into the yml file instead to keep it consistent with the rest of the repo.
> 
> See you already did this in https://github.com/OpenMined/PyDentity/pull/68/files#diff-232bac0e229b355a979026366e0a706bf87b48da75db720e309ed7fa1c6b8208 and https://github.com/OpenMined/PyDentity/pull/68/files#diff-540c821638d1899c5fb9b43372d6a108952362e99666e2514c333d4d66e54a61
> i.e. Not sure why it is required in the Dockerfile as well as part of the AcaPy startup cmd

The one in the docker command I probs used when trying to make it run first and didn't remove it when i moved it over to the yaml file. Will do that now.

**Comment by morrieinmaas** [comment]
@lohanspies shuld be the right way now

**Comment by lohanspies** [comment]
> @lohanspies shuld be the right way now

I am happy. Looked at the code changes and all make sense.
@wip-abramson ?

**Comment by wip-abramson** [comment]
Nice!

**Comment by wip-abramson** [comment]
I think this is all fine, but struggling to see the exact changes in the proofs.py file.

Did you check these changes do not affect any use of this API in the notebooks

**Comment by morrieinmaas** [comment]
> I think this is all fine, but struggling to see the exact changes in the proofs.py file.
> 
> Did you check these changes do not affect any use of this API in the notebooks

@wip-abramson I added `send_problem_report` and `send_presentation`. I re-ordered the functions in order they appear in the swagger API interface so it's easier to cross compare when working on it. I also checked whether any args have changed against swagger and that looks fine. I did have a quick run through the aries-basic-controller notebook and all worked (on my machine ;-P ). happy to have another run through though.

**Comment by wip-abramson** [comment]
No all good!

**Pull request merged**